### PR TITLE
Add summary MD to checks

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Adds a more user-friendly title and summary body to the checks.

Part of https://github.com/web-platform-tests/wpt.fyi/issues/712

## Review Information
@Hexcles - I suspect my golang "what dir am I in" code might not play well on appengine. Do you happen to know whether it will work?

### CODE REVIEW

**Comment by lukebjerring** [comment]
@foolip - fixing the circular dep problem in another PR (#747), but you can review what we want the contents of the check to be independent of that :)

File: shared/params.go [hunk 1]
```diff
@@ -141,6 +141,22 @@ func (p ProductSpec) Matches(run TestRun) bool {
 	return true
 }
 
+// DisplayName returns a capitalized version of the product's name.
```
Comment (foolip) [inline shared/params.go]: Do we have enough info to include the channel name here, or at the call site?

File: api/checks/api.go [hunk 2]
```diff
@@ -16,13 +18,13 @@ type SuitesAPI interface {
 	// a pending check_run for the given browser name for each CheckSuite.
 	// Returns true if any check_runs were created (i.e. any CheckSuite entities were
 	// found, and the create succeeded).
-	PendingCheckRun(sha, browser string) (bool, error)
+	PendingCheckRun(sha string, browser shared.ProductSpec) (bool, error)
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply api/checks/api.go]: Done: https://github.com/web-platform-tests/wpt.fyi/pull/751

File: api/checks/webhook.go [hunk 3]
```diff
@@ -266,15 +268,22 @@ func getSignedJWT(ctx context.Context) (string, error) {
 	return jwtToken.SignedString(key)
 }
 
-func getDetailsURL(ctx context.Context, sha, browser string) *url.URL {
-	hostname := shared.GetHostname(ctx)
-	detailsURL, _ := url.Parse(fmt.Sprintf("https://%s/results/", hostname))
+// Diff for the given product at master vs the given sha.
+func getMasterDiffURL(ctx context.Context, sha string, product shared.ProductSpec) *url.URL {
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply api/checks/webhook.go]: Let's discuss this in an issue instead; there's some implications in both directions.

The most-recent master run might be lagging behind when a PR lands (they're faster), so fixing to a value that's behind might not be desirable. Using the current SHA of master may also be a problem, if we aren't producing results for that SHA for some reason. And, as you suggest, having a rolling master might be bad.

File: api/checks/suites.go [hunk 4]
```diff
@@ -74,25 +76,42 @@ func pendingCheckRun(ctx context.Context, sha, browser string) (bool, error) {
...
 	} else if len(suites) < 1 {
 		return false, nil
 	}
-	detailsURL := getDetailsURL(ctx, sha, browser)
+	detailsURL := getMasterDiffURL(ctx, sha, product)
 	detailsURLStr := detailsURL.String()
 
+	host := shared.GetHostname(ctx)
+	completed := summaries.Completed{
+		HostName: "staging.wpt.fyi",
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply api/checks/suites.go]: Changed to `host`

File: api/checks/suites.go [hunk 5]
```diff
@@ -74,25 +76,42 @@ func pendingCheckRun(ctx context.Context, sha, browser string) (bool, error) {
...
+		HostURL:  fmt.Sprintf("https://%s/", host),
+		SHAURL:   getURL(ctx, shared.TestRunFilter{SHA: sha[:10]}).String(),
+		DiffURL:  getMasterDiffURL(ctx, sha, product).String(),
+	}
+	summary, err := completed.Compile()
+	if err != nil {
+		return false, err
+	}
+
+	title := fmt.Sprintf("wpt.fyi - %s results", product.DisplayName())
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply api/checks/suites.go]: It's set in the App, I think. In the main landing page of a PR, we don't see the suite title, just the run title(s), so the context might be nicer. An example for another run is 

> Taskcluster (pull_request)

File: api/checks/summaries/completed.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Results have successfully been scraped and added to [{{ .HostName }}]({{ .HostURL }}).
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply api/checks/summaries/completed.md]: Not yet; we don't have support for that param (and I'd probably go `?pr=1234`; the extra URL path handling cases isn't worth the extra headache).
It's currently just pointing to the homepage.

File: api/checks/summaries/completed.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+
+They can be compared to the latest `master` run here:
+{{ .DiffURL }}
+
+Or, to view _all_ the latest runs for the same revision:
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply api/checks/summaries/completed.md]: Neither; the SHA that was associated with the check_run (which was, at the time, the PR HeadSHA)

File: api/checks/summaries/pending.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+Results have been produced, and are being collected as we speak...
+
+They'll eventually be visible on [{{ .HostName }}]({{ .RunsURL }}).
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply api/checks/summaries/pending.md]: I don't think so. The intention is that the `/test-runs` UI will show pending runs eventually, whereas once the run has landed, we can link straight to it.

File: shared/params.go (hunk 1 above)
Comment (lukebjerring) [reply shared/params.go]: Yes, the channel is available in the label set. I'd do that in a separate PR, though.

**Comment by lukebjerring** [comment]
Ping @foolip

File: shared/params.go (hunk 1 above)
Comment (foolip) [reply shared/params.go]: SGTM.

File: api/checks/summaries/pending.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (foolip) [reply api/checks/summaries/pending.md]: Oh, so this just links to /runs?

File: api/checks/summaries/completed.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (foolip) [reply api/checks/summaries/completed.md]: That'd be what I called PR head revision. Point is given the context I thought "same revision" would probably refer to "master", just that doesn't quite make sense.

Perhaps best to just land the PR so it's possible to see this in action to see what the URLs all are :)

File: api/checks/summaries/completed.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (foolip) [reply api/checks/summaries/completed.md]: Alright, `?pr=1234` would be cool.

File: api/checks/suites.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (foolip) [reply api/checks/suites.go]: Just as well to get this deployed then I can see what strings show up where directly :)

File: api/checks/webhook.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (foolip) [reply api/checks/webhook.go]: I think the run we want would be one for the merge base if it exists, something as soon as possible after if it doesn't, and something as soon as possible before if there aren't any runs after yet. (Reason to prefer runs after the merge base is that the PR might build on something recent that affects the results of the tests being changed.)

But yeah, separate issue sounds good, or we'll just make sure to have two runs always for PRs and not have any logic for this :)

File: api/checks/summaries/pending.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (lukebjerring) [reply api/checks/summaries/pending.md]: That's right.

**Pull request merged**